packages/instantsearch-react/src/translatable.js

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+
+export default function translatable(defaultTranslations) {
+  return Composed => {
+    function Translatable(props) {
+      const {translations, ...otherProps} = props;
+      const translate = (key, ...params) => {
+        const translation =
+          translations && {}.hasOwnProperty.call(translations, key) ?
+            translations[key] :
+            defaultTranslations[key];
+        if (typeof translation === 'function') {
+          return translation(...params);
```

Comment:
Translations are defined as either a simple string `Search your website` or a function `pageNo => 'Page  ' + pageNo`. Here, we're testing whether the translation we've just retrieved on the above line is a function in order to apply it additional parameters.